common/environment/environment.cpp

```diff
@@ -1594,6 +1676,23 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByIndex(unsigned index) const
     return (CConstDropZoneInfo *) getCache(xpath.str());
 }
 
+IConstDfuQueueInfo * CLocalEnvironment::getDfuQueueByIndex(unsigned index) const
+{
+    if (!numOfDfuQueues || (index == 0))
+        return nullptr;
+
+    buildDfuQueueCache();
+    if (index > numOfDfuQueues)
+        return nullptr;
+
+    StringBuffer xpath("Software/DfuQueue[@id=\"");
+    xpath.append(index).append("\"]");
+    synchronized procedure(safeCache);
+    IConstDfuQueueInfo * retVal = (CConstDfuQueueInfo *) getCache(xpath.str());
+    return retVal;
```

Comment:
trivial: but cleaner as in other places as just:
```
    return (CConstDfuQueueInfo *) getCache(xpath.str());
```

Reply:
Changed (that was a debug solution only).